### PR TITLE
Refactor and Test solr update_work

### PR DESCRIPTION
**Pull request by hornc**
This PR takes @cdrini 's refactor and docstring additions and adds new tests I wrote to understand the behaviour of the update work scripts, and refactors the configuration methods now that the solr updater script runs in dev.

**TODO:**

- [x] Stub SOLR responses in tests for CI (Travis tests are currently failing)
- [x] Add docstrings for methods tested in `Test_update_items`

### CODE REVIEW

**Comment by hornc** [comment]
@mekarpeles , @cdrini 
Ok, let's get this merged! I think any further logging and fixes can be done in subsequent PRs. This was meant to document and test existing functionality, it's too big a task to fix everything in one go!

Thanks heaps for your commits and docstrings on this @cdrini !

File: openlibrary/tests/solr/test_update_work.py [hunk 1]
```diff
@@ -1,73 +1,101 @@
...
         d = build_data(work)
-        assert self.match_dicts(self.strip_empty_lists(d), {
-            "key": "/works/OL1M",
-            "title": "Foo",
-            "has_fulltext": False,
-            "edition_count": 0,
-        })
+        assert d["key"] == "/works/OL1M"
+        assert d["title"] == "Foo"
+        assert d["has_fulltext"] == False
```
Comment (mekarpeles) [inline openlibrary/tests/solr/test_update_work.py]: This should probably be done with an `is` comparison as opposed to `==`

File: openlibrary/tests/solr/test_update_work.py [hunk 2]
```diff
@@ -81,74 +109,114 @@ def test_edition_count(self):
...
-        e = make_edition(key="/books/OL1M", ocaid='foo00bar', _ia_meta={"collection":['lendinglibrary', 'americana']})
-        w = make_work(editions=[e])
+        w = make_work()
+        update_work.data_provider = FakeDataProvider([
+            w,
+            make_edition(w, key="/books/OL1M", ocaid='foo00bar',
+                         _ia_meta={"collection": ['lendinglibrary', 'americana']})
+        ])
         d = build_data(w)
         assert d['has_fulltext'] == True
```
Comment (mekarpeles) [inline openlibrary/tests/solr/test_update_work.py]: We can likely just `assert` without True, or assert `is True`

**Review by mekarpeles — APPROVED** [review]
few suggestions, no blockers; LGTM

File: openlibrary/tests/solr/test_update_work.py (hunk 2 above)
Comment (tfmorris) [reply openlibrary/tests/solr/test_update_work.py]: I actually prefer the explicit == True. It more explicitly declares the intent (and avoids any ambiguity about whether truthy values are acceptable).

**Pull request merged**